airflow/gcp/hooks/campaign_manager.py

```diff
@@ -0,0 +1,244 @@
+        delegate_to: Optional[str] = None,
+    ) -> None:
+        super().__init__(gcp_conn_id, delegate_to)
+        self.api_version = api_version
+
+    def get_conn(self) -> Resource:
+        """
+        Retrieves connection to Campaign Manager.
+        """
+        if not self._conn:
+            http_authorized = self._authorize()
+            self._conn = build(
+                "dfareporting",
+                self.api_version,
+                http=http_authorized,
+                cache_discovery=False,
+            )
+        return self._conn
+
+    def delete(self, profile_id: str, report_id: str) -> Any:
+        """
+        Deletes a report by its ID.
+
+        :param profile_id: The DFA user profile ID.
+        :type profile_id: str
+        :param report_id: The ID of the report.
+        :type report_id: str
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .delete(profileId=profile_id, reportId=report_id)
+            .execute(num_retries=self.num_retries)
+        )
+        return response
+
+    def insert(self, profile_id: str, report: Dict[str, Any]) -> Any:
+        """
+        Creates a report.
+
+        :param profile_id: The DFA user profile ID.
+        :type profile_id: str
+        :param report: The report resource to be inserted.
+        :type report: Dict[str, Any]
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .insert(profileId=profile_id, body=report)
+            .execute(num_retries=self.num_retries)
+        )
+        return response
+
+    def list(
+        self,
+        profile_id: str,
+        max_results: Optional[int] = None,
+        scope: Optional[str] = None,
+        sort_field: Optional[str] = None,
+        sort_order: Optional[str] = None,
+    ) -> list:
+        """
+        Retrieves list of reports.
+
+        :param profile_id: The DFA user profile ID.
+        :type profile_id: str
+        :param max_results: Maximum number of results to return.
+        :type max_results: Optional[int]
+        :param scope: The scope that defines which results are returned.
+        :type scope: Optional[str]
+        :param sort_field: The field by which to sort the list.
+        :type sort_field: Optional[str]
+        :param sort_order: Order of sorted results.
+        :type sort_order: Optional[str]
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .list(
+                profileId=profile_id,
+                maxResults=max_results,
+                scope=scope,
+                sortField=sort_field,
+                sortOrder=sort_order,
+            )
+            .execute(num_retries=self.num_retries)
+        )
+        reports = [r for r in response.get('items', [])]
+        next_token = response.get('nextPageToken')
```

Comment:
Is `list_next` method is supported by this API here?

Reply:
No information in API.

Reply:
It's a part of Google API Python Client . I think, you should confirm it manually. In my opinion, it should works.

Reply:
<img width="780" alt="Screenshot 2019-09-23 at 16 28 19" src="https://user-images.githubusercontent.com/12058428/65434495-353c2f80-de1f-11e9-9ca3-4582a6d0e368.png">

Reply:
Thanks for that! Indeed it works. Same was done in storage transfer service.